crates/asm/txs/bridge-v1/src/deposit_request/descriptor/format.rs

```diff
@@ -0,0 +1,355 @@
+//! Deposit descriptor wire format used by bridge deposits.
+//!
+//! A descriptor says "route this deposit to account serial X and subject bytes Y". The encoding is
+//! compact: it avoids zero padding in the subject and uses fewer bytes for small serial values.
+//!
+//! Layout (all values are big-endian):
+//! `[control:1][serial:1..=3][subject:0..=32]`
+//!
+//! Control byte (bit 7 is MSB):
+//! - bits 7..6: reserved, must be 0
+//! - bits 5..4: serial length (00=1 byte, 01=2 bytes, 10=3 bytes, 11 reserved)
+//! - bits 3..0: the 4 bits immediately above the encoded serial bytes
+//!
+//! Serial encoding:
+//! - A serial is a `u32`, but only 28 bits are encodable (max `0x0FFF_FFFF`).
+//! - Pick the smallest length that fits: `<= 0x0FFF` -> 1 byte, `<= 0x0FFFFF` -> 2 bytes, `<=
```

Comment:
>`<= 0x0FFF` -> 1 byte

as stated above, iiuc we are always forced to use 12 bits -- that's quite a lot don't you think?